.github/actions/verify-token/src/inputs.ts

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 SLSA Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WIHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as core from "@actions/core";
+import * as YAML from "yaml";
+import { rawTokenInterface, GitHubWorkflowInterface } from "../src/types";
+import { fetchToolWorkflow, asMap } from "./utils";
+
+export async function filterWorkflowInputs(
+  slsaToken: rawTokenInterface,
+  ghToken: string,
+  repoName: string,
+  hash: string,
+  workflowPath: string
+): Promise<rawTokenInterface> {
+  const content = await fetchToolWorkflow(
+    ghToken,
+    repoName,
+    hash,
+    workflowPath
+  );
+  return updateSLSAToken(content, slsaToken);
+}
+
+export function updateSLSAToken(
+  content: string,
+  slsaToken: rawTokenInterface
+): rawTokenInterface {
+  const ret = Object.create(slsaToken);
+  const workflow: GitHubWorkflowInterface = YAML.parse(content);
+  slsaToken.tool.inputs = asMap(slsaToken.tool.inputs);
```

Comment:
What is the type of `slsaToken.tool.inputs` initially?

Reply:
it's an `object`, even though we declare it as a `Map` in the structure

Reply:
hmm. I guess Map and Object are basically compatible types in TS. I'll need to look up the details.